### ISSUE TITLE
mount resource not idempotent for loopback mounts

### ISSUE THREAD

**Issue by Blorpy** [chef/chef#10918]
## Description

Since upgrading to chef client 16, from 15, the mount resource is no longer idempotent for loopback mounts. The resource tries to mount them every time, which causes converge to fail as the mount commands returns an error code indicating the mount already exists. 

## Chef Version

16.9.29

## Platform Version

CentOS 7

## Replication Case

```
mount "/var/www/mnt/disk1" do
  device "/mnt/public/disk1.iso"
  device_type :device
  options 'loop'
  action [:mount, :enable]
end
```

## Client Output

```
[2021-01-21T19:57:25+00:00] TRACE: Special device 0 mounted as /var/www/mnt/disk1
[2021-01-21T19:57:25+00:00] TRACE: Found mount /mnt/public/disk1.iso to /var/www/mnt/disk1 in /etc/fstab
mount: /mnt/public/disk1.iso is already mounted
```

**Comment by Blorpy** [comment]
```
(byebug) mounted?
false
(byebug) enabled?
true
```
Not sure how mounted? is returning false here for the resource in question...

**Comment by Blorpy** [comment]
I don't follow why mount would be set to false here when we have a match: https://github.com/chef/chef/blob/master/lib/chef/provider/mount/linux.rb#L50

**Comment by msys-sgarg** [comment]
Right now our logic to match the existing mounts withe mount specified in the recipe.. We have not considered the case for loop mounts. In case of loop mounts the output of command `findmnt -rn` will not contain the name of device. Hence it matches the regex - ` when %r{\A#{Regexp.escape(real_mount_point)}\s+([/\w])+\s}` and mounted is set to false just in case mount is of type loop.  - 
https://github.com/chef/chef/blob/master/lib/chef/provider/mount/linux.rb#L50

Below the example how the output of command looks like in case of loop mount 
`/var/www/mnt/disk1 /dev/loop0 iso9660 ro,relatime`

I have two options to provide :-
1) change `mounted = true` in case of mount point ( for ex /var/www/mnt/disk1 )being there in the output of `findmnt -rn`. May be we need to look into exactly why we kept it as false in the first place ? I do not see any issue happening when i change this flag to true.. I tested in two environments (Ubuntu and CentOs 7)

2) Create another scenario just to handle options of type `loop`. Problem here is that we would not know the mount option type in case no option is provided by user. `option defaults is different depending on the type of device`
